### PR TITLE
Update dependencies, remove retrolambda, fix tests, add kotlin

### PR DESCRIPTION
**Pull request by szymonkozak**

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -43,53 +42,53 @@ android {
...
+    implementation fileTree(dir: 'libs', include: ['*.jar'])
+
+    implementation "org.jetbrains.kotlin:kotlin-stdlib-jre7:$kotlin_version"
+    testImplementation 'junit:junit:4.12'
+    testImplementation 'org.mockito:mockito-core:1.10.9'
+
+    implementation 'com.google.dagger:dagger:2.14.1'
+    annotationProcessor 'com.google.dagger:dagger-compiler:2.14.1'
+
+    implementation 'com.android.support:design:27.0.2'
```
Comment (freszu) [inline app/build.gradle]: pull repeating version numbers to some const

File: app/build.gradle [hunk 2]
```diff
@@ -43,53 +42,53 @@ android {
...
+    implementation 'com.android.support:cardview-v7:27.0.2'
+
+    implementation 'me.zhanghai.android.materialprogressbar:library:1.4.2'
+    implementation 'org.greenrobot:eventbus:3.0.0'
+
+    implementation 'com.jakewharton:butterknife:8.0.1'
+    annotationProcessor 'com.jakewharton:butterknife-compiler:8.0.1'
```
Comment (freszu) [inline app/build.gradle]: as above

File: app/build.gradle [hunk 3]
```diff
@@ -43,53 +42,53 @@ android {
...
+    implementation 'com.afollestad.material-dialogs:core:0.9.6.0'
+    implementation 'com.davemorrissey.labs:subsampling-scale-image-view:3.6.0'
+
+    implementation 'io.reactivex:rxjava:1.2.9'
+    implementation 'io.reactivex:rxandroid:1.2.0'
+    implementation 'com.jakewharton.rxbinding:rxbinding:0.4.0'
+    implementation 'com.jakewharton.rxbinding:rxbinding-appcompat-v7:0.4.0'
+
+    implementation 'com.squareup.okhttp3:okhttp:3.9.1'
+    implementation 'com.squareup.okhttp3:logging-interceptor:3.9.1'
```
Comment (freszu) [inline app/build.gradle]: as above

File: app/build.gradle [hunk 4]
```diff
@@ -43,53 +42,53 @@ android {
...
+
+    implementation 'com.squareup.retrofit2:retrofit:2.3.0'
+    implementation 'com.squareup.retrofit2:adapter-rxjava:2.1.0'
```
Comment (freszu) [inline app/build.gradle]: why not 2.3.0?

File: app/build.gradle [hunk 5]
```diff
@@ -43,53 +42,53 @@ android {
...
+    implementation 'com.squareup.retrofit2:converter-gson:2.3.0'
```
Comment (freszu) [inline app/build.gradle]: and as above :D

File: app/build.gradle [hunk 6]
```diff
@@ -43,53 +42,53 @@ android {
...
+    debugImplementation 'com.squareup.leakcanary:leakcanary-android:1.5.1'
+    releaseImplementation 'com.squareup.leakcanary:leakcanary-android-no-op:1.5.1'
+    testImplementation 'com.squareup.leakcanary:leakcanary-android-no-op:1.5.1'
+    implementation 'com.google.code.gson:gson:2.8.2'
+    implementation 'com.squareup.sqlbrite:sqlbrite:0.6.3'
+    implementation 'net.sf.opencsv:opencsv:2.3'
+    implementation 'com.github.bumptech.glide:glide:3.8.0'
```
Comment (freszu) [inline app/build.gradle]: add anotation procesor for glide also

**Review by freszu — CHANGES_REQUESTED** [review]

File: build.gradle [hunk 7]
```diff
@@ -1,19 +1,21 @@
 // Top-level build file where you can add configuration options common to all sub-projects/modules.
 buildscript {
+    ext.kotlin_version = '1.2.21'
```
Comment (freszu) [inline build.gradle:3]: didn't knew about this notation

File: app/build.gradle (hunk 6 above)
Comment (szymonkozak) [reply app/build.gradle]: Glide < 4 doesn't require annotation processor.

File: app/build.gradle [hunk 8]
```diff
@@ -42,54 +41,64 @@ android {
...
+    implementation "com.github.bumptech.glide:glide:3.8.0"
+
+    implementation "com.google.android.gms:play-services-location:11.8.0"
+    implementation "com.google.android.gms:play-services-maps:11.8.0"
+    implementation "com.google.maps.android:android-maps-utils:0.5"
+    implementation "com.crashlytics.sdk.android:crashlytics:2.8.0"
 
-    compile 'com.vanniktech:onactivityresult:0.6.0'
-    apt 'com.vanniktech:onactivityresult-compiler:0.6.0'
+    implementation "com.vanniktech:onactivityresult:0.6.0"
```
Comment (freszu) [inline app/build.gradle]: also this

File: app/build.gradle [hunk 9]
```diff
@@ -42,54 +41,64 @@ android {
...
+    debugImplementation "com.squareup.leakcanary:leakcanary-android:$versions.leakcanary"
+    releaseImplementation "com.squareup.leakcanary:leakcanary-android-no-op:$versions.leakcanary"
+    testImplementation "com.squareup.leakcanary:leakcanary-android-no-op:$versions.leakcanary"
+    implementation "com.google.code.gson:gson:2.8.2"
+    implementation "com.squareup.sqlbrite:sqlbrite:0.6.3"
+    implementation "net.sf.opencsv:opencsv:2.3"
+    implementation "com.github.bumptech.glide:glide:3.8.0"
+
+    implementation "com.google.android.gms:play-services-location:11.8.0"
+    implementation "com.google.android.gms:play-services-maps:11.8.0"
```
Comment (freszu) [inline app/build.gradle]: this

**Review by freszu — APPROVED** [review]
only those two also and merge

**Pull request merged**